tests/extmod/machine_spi_rate.py

```diff
@@ -0,0 +1,128 @@
+# Test machine.SPI data transfer rates
+import sys
+
+
+try:
+    import time
+    from machine import Pin, SPI
+except ImportError:
+    print("SKIP")
+    raise SystemExit
+
+MAX_DELTA_MS = 8
+
+# Configure pins based on the board: tuples of spi_id, sck, mosi, miso
+if "pyboard" in sys.platform:
+    spi_instances = (
+        (1, None, None, None),  # "explicit choice of sck/mosi/miso is not implemented"
+        (2, None, None, None),
+    )
+elif "rp2" in sys.platform:
+    spi_instances = ((0, Pin(18), Pin(19), Pin(16)),)
+elif "esp32" in sys.platform:
+    spi_instances = [
+        (1, Pin(18), Pin(19), Pin(21)),
+    ]
+    if "ESP32C3" not in str(sys.implementation):
+        spi_instances.append((2, Pin(18), Pin(19), Pin(21)))
+else:
+    raise SystemExit("Please add support for this test on this platform.")
```

Comment:
I don't think this message ends up anywhere the user (person running the test) can see it.  It's not in the generated .out file.  The test just fails but the .out has no indication of why.  Probably better to make this an assert like `assert False, "platform not supported"`.

Reply:
Oh right, thanks - I don't know why I thought this worked on MP.

I went for "print and then SystemExit" as that's already a common pattern.